Source/Core/VideoBackends/D3D/D3DState.cpp

```diff
@@ -301,7 +301,14 @@ ID3D11SamplerState* StateCache::Get(SamplerState state)
   sampdc.MinLOD = state.min_lod / 16.f;
   sampdc.MipLODBias = (s32)state.lod_bias / 32.0f;
 
-  if (state.anisotropic_filtering)
+  //if texture is marked as hacky
+  if(state.lod_bias_scale)
+  {
+    auto offset = log2f(state.lod_bias_scale);
+    sampdc.MaxLOD += offset;
```

Comment:
Sorry, I haven't got why you modify the MaxLOD. The range of valid LOD values should not be altered here, so they should stay as they are.

Reply:
LOD is `min(lod + bias, max_lod)` so if you don't also offset the max lod, the values get clamped to something lower than needed. 
In practice, this breaks the water. I'm not totally sure of how exactly it works, but I can tell you I verified and several things set a max LOD in Mario Galaxy!